python/revng/profile/simple_daemon.py

```diff
@@ -0,0 +1,72 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import atexit
+import os
+import shutil
+import subprocess
+from signal import SIGINT, SIGKILL
+from time import sleep
+
+import requests
+
+from revng.scripting import DaemonProject
+
+from .base import Profile
+
+
+class SimpleDaemon(Profile):
+    def __init__(self, ip: str = "127.0.0.1", port: int = 8000, connection_retries: int = 10):
+        self.ip = ip
+        self.port = port
+        self.connection_retries = connection_retries
+        self.revng_executable = ""
+
+    def start_daemon(self, revng_executable: str = "", resume_dir: str = "") -> None:
+        def exit_handler(process):
+            process.send_signal(SIGINT)
+            try:
+                return process.wait(30.0)
+            except subprocess.TimeoutExpired:
+                process.send_signal(SIGKILL)
+
+            return process.wait()
+
+        self.revng_executable = revng_executable
+        if not self.revng_executable:
+            self.revng_executable = "revng"
+            assert shutil.which(self.revng_executable) is not None
+
+        env = os.environ
+        if resume_dir:
+            env["REVNG_DATA_DIR"] = resume_dir
+
+        cli_args = ["revng", "daemon", "-b", f"tcp:{self.ip}:{self.port}"]
+        daemon_process = subprocess.Popen(
+            cli_args, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT, env=env
+        )
+        atexit.register(exit_handler, daemon_process)
+
+    def stop_daemon(self):
+        atexit._run_exitfuncs()
+
+    def get_project(self) -> DaemonProject:
```

Comment:
The returned class should be a `LocalDaemonProject` which wraps the `DaemonProject` and implements `{start,stop}_daemon`. When calling `get_project` the `start_daemon` will be called. When `__del__` is called the `stop_daemon` is called, if necessary.

Reply:
Is the new [implementation](https://github.com/revng/revng/pull/398/commits/ed8f74270f974f9550032f5c0174440a4adb15c4#diff-3f5c67d8b968823595f37312e6a10820d91f5cdc9bd77d038dc436bf957a254c) as you intended?

Reply:
The implementation adds another layer of indirection, I'd avoid it. Rename `LocalDaemonProcess` into `LocalDaemonProject` and make it extend the `DaemonProject` class.